[WIP] JOSS paper (#1284)
TulipaEnergy/TulipaEnergyModel.jl: paper/paper.md

---
title: 'TulipaEnergyModel.jl: Energy system optimisation with fully flexible resolution'
tags:
  - Julia
  - energy-system optimisation
  - unit commitment
  - representative periods

>   - flexible resolution

Reviewer: ```suggestion
  - flexible temporal resolution
```

authors:
    - given-names: Abel
      surname: Soares Siqueira
      email: abel.siqueira@esciencecenter.nl
      affiliation: 1
      orcid: "https://orcid.org/0000-0003-4451-281X"
    - given-names: Diego A.
      surname: Tejada-Arango
      email: diego.tejadaarango@tno.nl
      affiliation: 2
      orcid: "https://orcid.org/0000-0002-3278-9283"
    - given-names: Germán
      surname: Morales-España
      email: german.morales@tno.nl

>       affiliation: 2

Reviewer: ```suggestion
      affiliation: 2,3
```

    - given-names: Lauren
      surname: Clisby
      email: lauren.clisby@tno.nl
      affiliation: 2
      orcid: "https://orcid.org/0009-0008-7848-4144"
    - given-names: Maaike
      surname: Elgersma
      email: m.b.elgersma@tudelft.nl
      affiliation: 3
    - given-names: Ni
      surname: Wang
      email: ni.wang@tno.nl
      affiliation: 2
      orcid: "https://orcid.org/0000-0001-7037-7004"
    - given-names: Suvayu
      surname: Ali
      email: s.ali@esciencecenter.nl
      affiliation: 1
    - given-names: Zhi
      surname: Gao
      email: z.gao1@uu.nl
      affiliation: 5
      orcid: "https://orcid.org/0000-0002-3817-8037"
affiliations:
  - name: Netherlands eScience Center, Netherlands
    ror: 00rbjv475
    index: 1
  - name: TNO - Netherlands Organisation for Applied Scientific Research, Netherlands
    ror: 01bnjb948
    index: 2
  - name: Delft University of Technology, Netherlands
    ror: 02e2c7k09
    index: 3
  - name: VTT Technical Research Centre of Finland, Netherlands
    ror: 04b181w54
    index: 4
  - name: Utrecht University, Netherlands
    ror: 04pp8hn57
    index: 5
date: 10 July 2025
bibliography: paper.bib
---

## Summary

> `TulipaEnergyModel.jl` is an optimization model for the electricity market that can be coupled with other energy sectors (e.g., hydrogen, heat, natural gas, etc.).

Reviewer: ```suggestion
`TulipaEnergyModel.jl` is an energy optimisation model for analysis of sector coupling (e.g., electricity, hydrogen, heat, natural gas, etc.).
```

> The optimization model determines the optimal investment and operation decisions for different types of assets (e.g., producers, consumers, conversion, storage, and transport).

Reviewer: ```suggestion
The optimisation model determines the optimal investment and operation decisions for different types of assets (e.g., producers, consumers, conversion, storage, and transport) over time.
```

Our main use case is modeling energy distribution in Europe, but there are no constraints preventing the user from extending to other use cases.

One of the main features of TulipaEnergyModel is that it accepts a _fully flexible resolution_ (cite) for the assets and flows.
In other words, the resolution at which the variables are defined don't have to be multiples of one another.
As a short example, consider the following example:

![Example of network with flexible resolution of assets and flows](images/flexible-time-resolution.png)

In the example, we look at 6 hours of a network. The flow between "H2" and "ccgt" has a resolution of 6 hours (i.e., the whole period), while from "ccgt" to the "balance", the resolution is 1 hour.
The resolution from "wind" to "phs" is 3 hours, and the resolution from "phs" to "balance" is not regular, starting with a 4 hours block and then a 2 hours block.
All these "time blocks" are handled by the Tulipa energy model to allow for more or less detailed solutions.
This implies that less variables and constraints are created, ensuring a faster solving speed, with little loss in accuracy.

> See (cite) for more details on the research behind fully flexible resolution.

Reviewer: ```suggestion
See [@Gao2025] for more details on the research behind fully flexible resolution.
```

> Another feature of the Tulipa model is the use of representative periods (cite), supported by another package in our ecosystem, `TulipaClustering.jl`.
> The representative periods are obtained by clustering the asset profiles (time series) from the full time frame to a much smaller one.
> For instance, instead of using a full year with 8760 hours (365 periods of 24 hours each), we can choose to have 30 representative periods of 24 hours each.
> These representative periods will be computed using `TulipaClustering.jl` so that each of the 365 original periods are replaced by a combination of the 30 representatives.
> Now, we can model within representative periods (30 periods of 24 hours each), and across periods (365 periods).
> Either way, we have a much smaller number of variables, making our model easier to solve.
> See (cite) for more details on the research behind representative periods.

Reviewer: ```suggestion
TulipaEnergyModel.jl is fundamentally focused on modeling innovation, with its mathematical formulations firmly rooted in scientific research. The model also includes other key features such as flexible connection of energy assets, seasonal storage modeling using representative periods, and tight formulations to prevent simultaneous charging and discharging. Furthermore, it incorporates ramping and unit commitment functionalities, along with the capability for multi-year investment planning.
```

> One main design choice of Tulipa is that data is passed (and kept) in a [DuckDB](https://duckdb.org) [@DuckDB] connection.

Reviewer: ```suggestion
### Software Design Innovations
One of the main software design choices for TulipaEnergyModel.jl is to maintain a [DuckDB](https://duckdb.org) [@DuckDB] connection from the input data to model creation and output generation.
```

This insight inspired the development of TulipaEnergyModel.jl, with the core philosophy of advancing the state-of-the-art in formulation quality by exploiting the following three strategies: 1) lowering computational cost while maintaining model fidelity, by representing the same problem with fewer constraints and variables [@Tejada2025], and by creating tighter mixed-integer programming (MIP) formulations [@MoralesEspana2013]. 2) increasing model fidelity without extra computational cost, e.g., by developing more accurate linear programming (LP) approximations for storage [@Elgersma2025] and other technologies [@gentile2016; @MoralesEspana2022]. Finally, 3) balancing computational burden with adaptive/flexible model fidelity, i.e., having different levels of detail in different parts of the model, in the temporal [@Gao2025], technological [@MoralesEspana2022] and spatial dimensions.

> These modelling strategies have significant computational benefits, especially when handling problems of realistic scale: with the spatial dimension of a continent, interacting with other energy carriers such as gas and hydrogen, and with a horizon of decades while reasoning about the (renewable) generation with an hourly resolution.

Reviewer 2: > while reasoning about the (renewable) generation with an hourly resolution.

I think this makes it a run-on sentence...

How about:
```suggestion
These modelling strategies have significant computational benefits, especially when handling problems of realistic scale: the spatial dimension of a continent; with electricity interacting with gas, hydrogen, and other energy carriers; optimising over a time horizon of decades, while maintaining hourly resolution for key aspects (i.e. renewable generation).
```
